scripts/build/esbuild-plugins/evaluate.mjs

```diff
@@ -1,35 +1,32 @@
 import createEsmUtils from "esm-utils";
+import serialize from "serialize-javascript";
 
 const { importModule } = createEsmUtils(import.meta);
 
 export default function esbuildPluginEvaluate() {
   return {
     name: "evaluate",
-
     setup(build) {
-      build.onLoad({ filter: /\.evaluate\.c?js$/ }, async ({ path }) => {
-        let data = await importModule(path);
+      const { format } = build.initialOptions;
 
-        if (Object.hasOwn(data, "default")) {
-          if (Object.keys(data).length !== 1) {
-            throw new TypeError("Mixed export not allowed.");
-          }
-
-          data = data.default;
-        }
+      build.onLoad({ filter: /\.evaluate\.c?js$/ }, async ({ path }) => {
+        const module = await importModule(path);
+        const text = Object.entries(module)
+          .map(([key, value]) => {
```

Comment:
I think it is better to validate that `key` is a string. There is no problem in ESM. However, CJS can export the module like below:

```js
"use strict";

export[{}] = "foo":
```

This will be evaluated as `{ '[object Object]': 'foo' }`.
I know this kind of code is rarely written, but just in case.

Reply:
https://github.com/prettier/prettier/pull/13724/commits/0df9a767648061c688b883a09a81c4b8f91c7d42